### ISSUE TITLE
ERROR: unable to download video data: HTTP Error 403: Forbidden

### ISSUE THREAD

**Issue by linxman** [yt-dlp/yt-dlp#11876]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

USA/Virginia

### Provide a description that is worded well enough to be understood

Yesterday morning, I discovered that in trying to download a youtube video I encountered the following error:

ERROR: unable to download video data: HTTP Error 403: Forbidden

This error has occurred with all youtube videos that I have tried since then.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ yt-dlp -vU https://www.youtube.com/watch?v=Nu24kbhPRPo
[debug] Command-line config: ['-vU', 'https://www.youtube.com/watch?v=Nu24kbhPRPo']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.12.13 from yt-dlp/yt-dlp [542166962] (debian*)
[debug] Python 3.12.8 (CPython x86_64 64bit) - Linux-6.12.5-amd64-x86_64-with-glibc2.40 (OpenSSL 3.3.2 3 Sep 2024, glibc 2.40)
[debug] exe versions: ffmpeg 6.1.1-5 (setts), ffprobe 6.1.1-5, phantomjs 2.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.08.30, mutagen-1.47.0, pyxattr-0.8.1, requests-2.32.3, sqlite3-3.46.1, urllib3-2.2.3, websockets-14.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1837 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.12.13 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.12.13 from yt-dlp/yt-dlp)
[youtube] Extracting URL: https://www.youtube.com/watch?v=Nu24kbhPRPo
[youtube] Nu24kbhPRPo: Downloading webpage
[youtube] Nu24kbhPRPo: Downloading ios player API JSON
[youtube] Nu24kbhPRPo: Downloading mweb player API JSON
[debug] Loading youtube-nsig.03dbdfab from cache
[debug] [youtube] Decrypted nsig GAFaRrvQFKNwThYM7 => WUyMymkH2BPH-A
[debug] Loading youtube-nsig.03dbdfab from cache
[debug] [youtube] Decrypted nsig i9zzJuICuCs0hrUvw => 1-qed7Dg6-mh2A
[youtube] Nu24kbhPRPo: Downloading m3u8 information
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec, channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] Testing format 616
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 227
[download] Destination: /tmp/tmp_rl8yq92.tmp
[download] Got error: HTTP Error 403: Forbidden
  File "/bin/yt-dlp", line 8, in <module>
    sys.exit(main())
  File "/usr/lib/python3/dist-packages/yt_dlp/__init__.py", line 1093, in main
    _exit(*variadic(_real_main(argv)))
  File "/usr/lib/python3/dist-packages/yt_dlp/__init__.py", line 1083, in _real_main
    return ydl.download(all_urls)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 3605, in download
    self.__download_wrapper(self.extract_info)(
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 3578, in wrapper
    res = func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1613, in extract_info
    return self.__extract_info(url, self.get_info_extractor(key), download, extra_info, process)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1624, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1780, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1839, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2965, in process_video_result
    formats_to_download = self._select_formats(formats, format_selector)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2189, in _select_formats
    return list(selector({
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2428, in selector_function
    yield from f(ctx)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2527, in final_selector
    return selector_function(ctx_copy)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2439, in selector_function
    picked_formats = list(f(ctx))
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2448, in selector_function
    for pair in itertools.product(selector_1(ctx), selector_2(ctx)):
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2517, in selector_function
    yield matches[format_idx - 1]
  File "/usr/lib/python3/dist-packages/yt_dlp/utils/_utils.py", line 2252, in __getitem__
    self._cache.extend(itertools.islice(self._iterable, n))
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2418, in _check_formats
    yield from self._check_formats([f])
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2173, in _check_formats
    success, _ = self.dl(temp_file.name, f, test=True)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 3199, in dl
    return fd.download(name, new_info, subtitle)
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/common.py", line 464, in download
    ret = self.real_download(filename, info_dict)
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/hls.py", line 381, in real_download
    return self.download_and_append_fragments(ctx, fragments, info_dict)
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/fragment.py", line 513, in download_and_append_fragments
    download_fragment(fragment, ctx)
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/fragment.py", line 459, in download_fragment
    for retry in RetryManager(self.params.get('fragment_retries'), error_callback):
  File "/usr/lib/python3/dist-packages/yt_dlp/utils/_utils.py", line 5251, in __iter__
    self.error_callback(self.error, self.attempt, self.retries)
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/fragment.py", line 456, in error_callback
    self.report_retry(err, count, retries, frag_index, fatal)
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/common.py", line 410, in report_retry
    RetryManager.report_retry(
  File "/usr/lib/python3/dist-packages/yt_dlp/utils/_utils.py", line 5258, in report_retry
    return error(f'{e}. Giving up after {count - 1} retries') if count > 1 else error(str(e))
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/common.py", line 413, in <lambda>
    error=IDENTITY if not fatal else lambda e: self.report_error(f'\r[download] Got error: {e}'),
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1090, in report_error
    self.trouble(f'{self._format_err("ERROR:", self.Styles.ERROR)} {message}', *args, **kwargs)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1018, in trouble
    tb_data = traceback.format_list(traceback.extract_stack())

ERROR: fragment 1 not found, unable to continue
  File "/bin/yt-dlp", line 8, in <module>
    sys.exit(main())
  File "/usr/lib/python3/dist-packages/yt_dlp/__init__.py", line 1093, in main
    _exit(*variadic(_real_main(argv)))
  File "/usr/lib/python3/dist-packages/yt_dlp/__init__.py", line 1083, in _real_main
    return ydl.download(all_urls)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 3605, in download
    self.__download_wrapper(self.extract_info)(
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 3578, in wrapper
    res = func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1613, in extract_info
    return self.__extract_info(url, self.get_info_extractor(key), download, extra_info, process)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1624, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1780, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1839, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2965, in process_video_result
    formats_to_download = self._select_formats(formats, format_selector)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2189, in _select_formats
    return list(selector({
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2428, in selector_function
    yield from f(ctx)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2527, in final_selector
    return selector_function(ctx_copy)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2439, in selector_function
    picked_formats = list(f(ctx))
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2448, in selector_function
    for pair in itertools.product(selector_1(ctx), selector_2(ctx)):
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2517, in selector_function
    yield matches[format_idx - 1]
  File "/usr/lib/python3/dist-packages/yt_dlp/utils/_utils.py", line 2252, in __getitem__
    self._cache.extend(itertools.islice(self._iterable, n))
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2418, in _check_formats
    yield from self._check_formats([f])
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 2173, in _check_formats
    success, _ = self.dl(temp_file.name, f, test=True)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 3199, in dl
    return fd.download(name, new_info, subtitle)
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/common.py", line 464, in download
    ret = self.real_download(filename, info_dict)
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/hls.py", line 381, in real_download
    return self.download_and_append_fragments(ctx, fragments, info_dict)
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/fragment.py", line 514, in download_and_append_fragments
    result = append_fragment(
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/fragment.py", line 479, in append_fragment
    self.report_error(f'fragment {frag_index} not found, unable to continue')
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1090, in report_error
    self.trouble(f'{self._format_err("ERROR:", self.Styles.ERROR)} {message}', *args, **kwargs)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1018, in trouble
    tb_data = traceback.format_list(traceback.extract_stack())


[info] Unable to download format 616. Skipping...
[info] Nu24kbhPRPo: Downloading 1 format(s): 248+251
[debug] Invoking http downloader on "https://rr5---sn-5ualdnss.googlevideo.com/videoplayback?expire=1734910224&ei=sExoZ9rON9Oty_sP-tCp4Qs&ip=173.162.21.75&id=o-AFSsk061Ia8wWkIIPAE4P339yEpjph0-C1gRzEmHwkFA&itag=248&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&met=1734888624%2C&mh=Dp&mm=31%2C29&mn=sn-5ualdnss%2Csn-5uaezned&ms=au%2Crdu&mv=m&mvi=5&pl=20&rms=au%2Cau&initcwndbps=4213750&bui=AfMhrI-ClE9FN4m6rN9IuQq894v_R4qNQ0GbVnqSxvFE4nOf2STNrfYhwX5L10JowMbHcc9AcayRkGbF&spc=x-caUHiPZ2v7mOig-zs8fD6Dilqu1T9rs9RrDh2QKsDMnDuZeAxB&vprv=1&svpuc=1&mime=video%2Fwebm&rqh=1&gir=yes&clen=128830112&dur=994.999&lmt=1734759288811738&mt=1734888314&fvip=5&keepalive=yes&fexp=51326932%2C51335594%2C51371294&c=IOS&txp=330F224&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIhAIuy6hb9t8RxJz_Xefh3PEoFeg1RvdOF9dZ7b5v5XM_ZAiBvMMGvl3rh3IfPKKnrLA3hLycKtU4zM1u2fH3tb0hJ4w%3D%3D&lsparams=met%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=AGluJ3MwRQIhALC7Hm1ByzHseuGkHr5gB-LsMOF7yZVaYIJdjTLk8HPSAiAxw7gi40CYFZilzkkhh0VyhEy83xk9vAVhvY2SzY4jWg%3D%3D"
ERROR: unable to download video data: HTTP Error 403: Forbidden
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 3461, in process_info
    partial_success, real_download = self.dl(fname, new_info)
                                     ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 3199, in dl
    return fd.download(name, new_info, subtitle)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/common.py", line 464, in download
    ret = self.real_download(filename, info_dict)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/http.py", line 367, in real_download
    establish_connection()
  File "/usr/lib/python3/dist-packages/yt_dlp/downloader/http.py", line 118, in establish_connection
    ctx.data = self.ydl.urlopen(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 4162, in urlopen
    return self._request_director.send(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/yt_dlp/networking/common.py", line 117, in send
    response = handler.send(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/yt_dlp/networking/_helper.py", line 208, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/yt_dlp/networking/common.py", line 340, in send
    return self._send(request)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/yt_dlp/networking/_requests.py", line 365, in _send
    raise HTTPError(res, redirect_loop=max_redirects_exceeded)
yt_dlp.networking.exceptions.HTTPError: HTTP Error 403: Forbidden
```

**Comment by bashonly** [comment]
I know I had asked you to open a proper issue for this, and I appreciate that you did, but someone else beat you to it. Closing as duplicate of #11868

**Comment by Muczak** [comment]
Issue:
ERROR: unable to download video data: HTTP Error 403: Forbidden

What’s happening:
YouTube (and some other sites) are blocking requests because they detect yt-dlp as an automated client instead of a regular browser.
This can happen due to missing authentication cookies, invalid headers, or updated site anti-bot checks.
The yt-dlp team has already marked similar cases as “site-bug”.

Fix that worked ✅

Open YouTube in your regular browser and make sure you’re logged in.

Install the extension “Get cookies.txt LOCALLY” (for Chrome/Edge/Firefox).

Export your YouTube cookies as a file named cookies.txt.

Place that cookies.txt in the same folder where yt-dlp or your bot runs.

Run yt-dlp with the cookies file, for example:

yt-dlp --cookies cookies.txt "https://www.youtube.com/watch?v=VIDEO_ID"


Check that the download starts correctly (it should no longer show HTTP 403).

Why this works:
Using cookies makes your yt-dlp requests look exactly like a logged-in browser session, bypassing YouTube’s stricter anti-bot filtering that was recently introduced.

Additional notes:

Always update yt-dlp to the latest version (yt-dlp -U).

If running from a server, make sure its IP isn’t rate-limited or region-blocked by YouTube.

If even cookies don’t fix it, the issue is likely on YouTube’s side and requires a new extractor update in yt-dlp.

Summary:
If you get a 403 error starting around October 2025, downloading your YouTube cookies and adding
--cookies cookies.txt resolves it instantly.
This workaround is confirmed working until yt-dlp releases an official patch for the new YouTube request validation.
